main/src/main/scala/sbt/Cross.scala

```diff
@@ -216,12 +216,30 @@ object Cross {
     Command.arb(requireSession(switchParser), switchHelp)(switchCommandImpl)
 
   private def switchCommandImpl(state: State, args: Switch): State = {
-    val switchedState = switchScalaVersion(args, state)
+    val x = Project.extract(state)
+    val (switchedState, affectedRefs) = switchScalaVersion(args, state)
+
+    val strictCmd =
+      if (args.version.force) {
+        // The Scala version was forced on the whole build, run as is
+        args.command
+      } else {
+        args.command.map { rawCmd =>
+          parseCommand(rawCmd) match {
+            case Right(_) => rawCmd // A project is specified, run as is
+            case Left(cmd) =>
+              resolveAggregates(x)
+                .intersect(affectedRefs)
+                .collect { case ProjectRef(_, proj) => s"$proj/$cmd" }
+                .mkString("all ", " ", "")
```

Comment:
I _think_ this might have broken `+ -v`:

```
sbt> + -v compile
[error] Expected ';'
[error] ++ -v 2.11.12! all client/Compile/compile offline/Compile/compile example-offline/Compile/compile registry/Compile/compile online/Compile/compile
```

Thanks for working on this issue back when, `++` is behaving itself. Now sbt still seems to need work on `+`, it _maybe_ does the right thing for me with the explicit `crossScalaVersions := Nil` on the root aggregating project. Seems brittle, and esoteric…